File: src/PrimeNumberOfSetBitsInBinaryRepresentation.java
public class PrimeNumberOfSetBitsInBinaryRepresentation {
    private final byte[] isPrime = new byte[] {0, 0, 1, 1, 0, 1, 0,
            1, 0, 0, 0, 1, 0,
            1, 0, 0, 0, 1, 0,
            1, 0, 0, 0, 1, 0
    };

    public int countPrimeSetBits(int left, int right) {
        int primeSetBits = 0;
        for (int number = left ; number <= right ; number++) {
            if (isPrime[setBits(number)] == 1) primeSetBits++;
        }
        return primeSetBits;
    }

    private int setBits(int number) {
        int setBits = 0;
        while (number > 0) {
            setBits += number & 1;
            number >>= 1;
        }
        return setBits;
    }
}

File: python/prime_number_of_set_bits_in_binary_representation.py
class Solution:
    def __init__(self):
        self.isPrime = [
            0, 0, 1, 1, 0, 1, 0,
            1, 0, 0, 0, 1, 0,
            1, 0, 0, 0, 1, 0,
            1, 0, 0, 0, 1, 0
        ]

    def countPrimeSetBits(self, left: int, right: int) -> int:
        primeSetBits = 0
        for number in range(left, right + 1):
            if self.isPrime[self.setBits(number)] == 1:
                primeSetBits += 1
        return primeSetBits

    def setBits(self, number: int) -> int:
        result = 0
        while number > 0:
            result += number & 1
            number >>= 1
        return result
